### PR TITLE
[LIBCLOUD-751] Add tests in Point DNS provider for Redirects and Mail…

### PR DESCRIPTION
**Pull request by aleGpereira**
…Redirects classes

### CODE REVIEW

**Comment by Kami** [comment]
I made a change to the `PointDNSException` class so it inherits from `ProviderError` which is intended to be used for that purpose - 08556656a7d69ad3d5c696f14d8d41b95373c5ee. I also updated the constructor signature so it matches the parent one.

For the future reference - this class doesn't really provide additional value over the existing `ProviderError` one (all the information is already available via ProviderError class) so there is little value in adding / having it.

Besides that the changes look good so I've merged it into trunk.

Thanks.

File: libcloud/dns/drivers/pointdns.py [hunk 1]
```diff
@@ -332,9 +371,17 @@ def update_record(self, record, name, type, data, extra=None):
...
-                                           (zone.id, record.id),
-                                           method='PUT', data=r_data)
+        try:
+            response = self.connection.request('/zones/%s/records/%s' %
+                                               (zone.id, record.id),
+                                               method='PUT', data=r_data)
+        except (BaseHTTPError, MalformedResponseError):
+            e = sys.exc_info()[1]
+            if isinstance(e, MalformedResponseError) and e.body == 'Not found':
+                raise RecordDoesNotExistError(value="Record doesn't exists",
```
Comment (Kami) [inline libcloud/dns/drivers/pointdns.py:381]: Another "for the future reference" - Ideally those exceptions would be handled on the Response class - see how other classes use `set_context` method.

**Comment by aleGpereira** [comment]
Yeah, you're right. My intention only was to specify the provider. If for some reason someone needs to catch an exception only for pointdns, can use it. But, yeah. Is just a little thing.

**Comment by aleGpereira** [comment]
Thanks for the merges!

**Pull request closed without merging**